src/VoteDelegate.sol

```diff
@@ -50,6 +51,11 @@ contract VoteDelegate {
     ChiefLike   immutable public chief;
     PollingLike immutable public polling;
 
+    // --- constants ---
+
+    uint256 public constant HATCH_SIZE     = 5;
+    uint256 public constant HATCH_COOLDOWN = 5;
```

Comment:
I'm a little on the fence as to whether these are the right length or not. Gut feeling is that they're close but a couple more blocks for each wouldn't hurt (especially for the cooldown, as that timescale is one a human as opposed to a bot is more likely to have to deal with--one minute to do a tx isn't much, especially if you're carefully checking details etc). That said these values are probably okay and you certainly don't want to make them too long.

Reply:
the way i see it, cooldown is for mitigating denial of service attack against users who want to delegate. And 5 blocks seems to short imo. The way such an attack will happen is that a whale will call `lock`, and the attacker will call `reserveHatch` when the whale's tx is pending.
Then after the whale tx reverts, he will have at most 5 more blocks (~1 minute) to submit a new tx (and for it to get mined) before the attacker can call `reserveHatch` again.
And 1 minute is very short for a user who does manual txs (in particular when it can easily take almost a minute for the tx to get mined).